### PR TITLE
Acl implement widgets

### PR DESCRIPTION
**Pull request by nikitsenka**

### CODE REVIEW

**Comment by nikitsenka** [comment]
I see different styles in many services. 
setter - https://github.com/reportportal/service-api/blob/develop/src/main/java/com/epam/ta/reportportal/core/widget/impl/GetWidgetHandler.java
constructor - https://github.com/reportportal/service-api/blob/develop/src/main/java/com/epam/ta/reportportal/ws/controller/impl/ProjectController.java
field - https://github.com/reportportal/service-api/blob/develop/src/main/java/com/epam/ta/reportportal/ws/controller/impl/DashboardController.java
I prefer to use Autowired field for better readbility.

File: src/main/java/com/epam/ta/reportportal/core/widget/IShareWidgetHandler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.ta.reportportal.core.widget;
+
+import com.epam.ta.reportportal.entity.widget.Widget;
+
+public interface IShareWidgetHandler {
```
[the remark this answers is not in the document]
Comment (nikitsenka) [reply src/main/java/com/epam/ta/reportportal/core/widget/IShareWidgetHandler.java]: may i do it for all(widget, filter, dashboard) handlers in separate refactoring PR?

[the remark this answers is not in the document]
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/core/widget/IShareWidgetHandler.java]: > I see different styles in many services.
> setter - https://github.com/reportportal/service-api/blob/develop/src/main/java/com/epam/ta/reportportal/core/widget/impl/GetWidgetHandler.java
> constructor - https://github.com/reportportal/service-api/blob/develop/src/main/java/com/epam/ta/reportportal/ws/controller/impl/ProjectController.java
> field - https://github.com/reportportal/service-api/blob/develop/src/main/java/com/epam/ta/reportportal/ws/controller/impl/DashboardController.java
> I prefer to use Autowired field for better readbility.

If it is autowired by field so how to make a beautiful inject of mocks of those services during testing then?

[the remark this answers is not in the document]
Comment (nikitsenka) [reply src/main/java/com/epam/ta/reportportal/core/widget/IShareWidgetHandler.java]: https://github.com/reportportal/service-api/pull/415

[the remark this answers is not in the document]
Comment (nikitsenka) [reply src/main/java/com/epam/ta/reportportal/core/widget/IShareWidgetHandler.java]: >let's get rid of "I", rename to ShareWidgetHandler
Fixed for both widget and user filter.

**Pull request merged**